src/features/Chat/components/ActiveWindow/index.tsx

```diff
@@ -0,0 +1,234 @@
+import { selectUserId } from '@/features/Authentication/userSlice';
+
+import { useTranslation } from 'react-i18next';
+
+import styled from 'styled-components';
+import { palette } from '@/components/variables';
+import { Profile as ProfileIcon } from '@/components/Icons/Profile';
+import ArchivedIcon from '@/static/icons/archived-chats.svg';
+import BlockedIcon from '@/static/icons/blocked-chats.svg';
+import Text from '@/components/Text';
+import TextInput from '@/components/TextInput';
+import { IconButton, StatusButton } from '@/components/Buttons';
+import { MessageList } from './MessageList';
+
+const searchInputIconSize = 24;
+const closeInputIconSize = 34;
+
+const ActiveWindow = () => {
+  const { t } = useTranslation('chat');
+  const [showSearch, setShowSearch] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [text, setText] = useState('');
+  const [sendMessage, { isLoading: isMessageSendLoading }] =
+    useSendMessageMutation();
+  const [updateChatStatus] = useUpdateStatusMutation();
+  const userId = useAppSelector(selectUserId);
+
+  const chat = useAppSelector(selectActiveChat);
+  const isLoadingMessages = useAppSelector(
+    selectIsLoadingBuddyMessages(chat?.buddyId),
+  );
+
+  const updateStatus = (status: ChatFolder) => {
+    if (!userId || !chat) return;
+    updateChatStatus({ userId, buddyId: chat.buddyId, status });
+  };
+
+  const handleMessageSend = (buddyId: string, text: string) => {
+    if (!userId || isMessageSendLoading) return;
+
+    const message = toSendMessage(buddyId, userId, text);
+    sendMessage({ userId, message });
+    setText('');
+  };
+
+  const isLoading = isLoadingMessages || isMessageSendLoading;
+
+  const getProfileIcon = (status: ChatFolder) => {
+    const icons = {
+      ok: <ProfileIcon color="purpleDark" />,
+      archived: <img src={ArchivedIcon} />,
+      banned: <img src={BlockedIcon} />,
+    };
+
+    return icons[status];
+  };
+
+  return (
+    chat && (
+      <ActiveChatContainer>
+        <HeaderBar>
+          <ProfileInfo>
+            {getProfileIcon(chat.status)}
+            <MentorName variant="h2">{chat.displayName}</MentorName>
+            <MentorBio variant="p">{chat.status}</MentorBio>
+          </ProfileInfo>
+          {showSearch ? (
+            <SearchBar>
+              <SearchInput
+                variant="iconInput"
+                color={searchValue ? 'blueDark' : 'greyFaded'}
+                leftIcon={{
+                  sizeInPx: searchInputIconSize,
+                  variant: 'search',
+                }}
+                rightButton={{
+                  onClick: () => setShowSearch(false),
+                  sizeInPx: closeInputIconSize,
+                  variant: 'closeWithBackground',
+                }}
+                onChange={setSearchValue}
+                placeholder={t('header.search')}
+                value={searchValue}
+              />
+            </SearchBar>
```

Comment:
Maybe for the next PR: This could be abstracted to own file where the showSearch-is contains the rendering logic. It would clean this file a bit from unnecessary stuff

Reply:
Yes this could be!